compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -2163,4 +2167,327 @@ LogicalResult OnlineAttentionOp::getResultTilePosition(
+  SmallVector<OpFoldResult> lbs =
+      linalg::computeTileOffsets(builder, loc, ivs, tileSizes);
+  SmallVector<OpFoldResult> sizeBounds = llvm::to_vector(tileSizes);
+
+  std::tie(lbs, sizeBounds) = appendOffsetAndSizeForSymbolDimensions(
+      builder, customOp, lbs, sizeBounds);
+
+  tileSizes.resize(tileSizes.size() + numSymbols, builder.getIndexAttr(0));
+  SmallVector<OpFoldResult> subShapeSizes =
+      linalg::computeTileSizes(builder, loc, tileSizes, sizeBounds);
+
+  SmallVector<AffineExpr> symbolReplacements;
+  if (numSymbols != 0) {
+    symbolReplacements =
+        getDimExprsForSymbols(builder.getContext(), numDims, numSymbols);
+  }
+
+  assert(static_cast<int64_t>(valuesToTile.size()) <=
+             customOp->getNumOperands() &&
+         "more value to tile than operands.");
+  SmallVector<std::optional<linalg::SliceParameters>> allSliceParams;
+  allSliceParams.reserve(valuesToTile.size());
+  for (auto [opOperand, val] :
+       llvm::zip(customOp->getOpOperands(), valuesToTile)) {
+    Value shapedOp = val;
+    LLVM_DEBUG(llvm::dbgs() << "makeTiledShapes: for operand " << shapedOp);
+    AffineMap map = customOp.getMatchingIndexingMap(&opOperand);
+    if (numSymbols != 0) {
+      map = convertDimsToSymbols(map, numDims, numSymbols, symbolReplacements);
+    }
+
+    // If the map is empty, we dont tile this operand.
+    Type operandType = opOperand.get().getType();
+    if (map.isEmpty() || !isa<ShapedType>(operandType)) {
+      allSliceParams.push_back(std::nullopt);
+      LLVM_DEBUG(llvm::dbgs()
+                 << ": not tiled: use shape: " << operandType << "\n");
+      continue;
+    }
+
+    allSliceParams.push_back(linalg::computeSliceParameters(
+        builder, loc, shapedOp, tileSizes, map, lbs, sizeBounds, subShapeSizes,
+        /*omitPartialTileCheck=*/true));
+  }
+
+  return allSliceParams;
+}
+
+/// This method is same as `materializeTiledShape` method defined in
+/// `mlir/Dialect/Linalg/Utils/Utils.[h|cpp]`.
+static Operation *
+materializeTiledShape(OpBuilder &builder, Location loc, Value valueToTile,
+                      const linalg::SliceParameters &sliceParams) {
+  auto shapedType = dyn_cast<ShapedType>(valueToTile.getType());
+  auto *sliceOp = TypeSwitch<ShapedType, Operation *>(shapedType)
+                      .Case([&](RankedTensorType) {
+                        return builder.create<tensor::ExtractSliceOp>(
+                            loc, valueToTile, sliceParams.offsets,
+                            sliceParams.sizes, sliceParams.strides);
+                      })
+                      .Default([](ShapedType) -> Operation * {
+                        llvm_unreachable("Unexpected shaped type");
+                      });
+  return sliceOp;
+}
+
+/// This method is adapted from the method `linalg::makeTiledShapes`,
+/// adapted to work with map that have symbols, and empty maps.
+SmallVector<Value> makeCustomOpTiledShapes(OpBuilder &builder, Location loc,
```

Comment:
nit: add `static` keyword.

```suggestion
static SmallVector<Value> makeCustomOpTiledShapes(OpBuilder &builder, Location loc,
```